### PR TITLE
fix of two vertical bands overlaying

### PR DESCRIPTION
**Pull request by rushan-zagidullin**
Sample. This solution works for my situation discribed in [CUBA forum](https://www.cuba-platform.ru/support/topic/%D0%B2%D0%BE%D0%BF%D1%80%D0%BE%D1%81-%D0%BF%D0%BE-%D0%BF%D0%BE%D0%B2%D0%BE%D0%B4%D1%83-%D0%B3%D0%B5%D0%BD%D0%B5%D1%80%D0%B0%D1%82%D0%BE%D1%80%D0%B0-%D0%BE%D1%82%D1%87%D0%B5%D1%82%D0%BE%D0%B2-yarg) This solution is not fully tested

### CODE REVIEW

**Comment by ODE-SA** [comment]
tried this fix and it does not fix our issue (1 vertical band that extends to overlap a previous horizontal band). LibreOffice opens the file correctly, M$Excel gives warning about errors, then when asked to fix them the whole place that was supposed to be populated by the vertical band is empty.

If the horizontal band is defined (and populated) after the vertical band expansion, then M$Excel works correctly.

File: core/modules/core/src/com/haulmont/yarg/formatters/impl/XlsxFormatter.java [hunk 1]
```diff
@@ -591,6 +602,17 @@ protected Row findNextRowForVBand(BandData band, Range templateRange, List<Row>
         return firstRow;
     }
 
+    protected boolean isNotRenderedYet(List<BandData> bandDataList) {
+        if(CollectionUtils.isNotEmpty(bandDataList)) {
+            for (BandData bandData : bandDataList) {
```
Comment (belyaev-andrey) [inline core/modules/core/src/com/haulmont/yarg/formatters/impl/XlsxFormatter.java]: Can we use stream.containsAny here?

File: core/modules/core/src/com/haulmont/yarg/formatters/impl/XlsxFormatter.java [hunk 2]
```diff
@@ -573,7 +573,18 @@ protected Row findNextRowForHBand(BandData band, Range templateRange, List<Row>
     protected Row findNextRowForVBand(BandData band, Range templateRange, List<Row> resultSheetRows) {
         Row firstRow = null;
         boolean isFirstLevelBand = BandData.ROOT_BAND_NAME.equals(band.getParentBand().getName());
-        previousRangesRightOffset = 0;
+        //if this band is first rendered band of all its siblings
+        if (isNotRenderedYet(band.getParentBand().getChildrenList())) {
```
Comment (belyaev-andrey) [inline core/modules/core/src/com/haulmont/yarg/formatters/impl/XlsxFormatter.java]: You can get NPE here because you check it later.

**Comment by kkorada** [comment]
is this issue fixed ?

**Pull request closed without merging**